save-plugins/warn-plugin/src/commonMain/kotlin/org/cqfn/save/plugin/warn/utils/Warning.kt

```diff
@@ -108,16 +110,22 @@ internal fun String.extractWarning(warningRegex: Regex,
     "SwallowedException")
 internal fun String.getLineNumber(warningRegex: Regex,
                                   lineGroupIdx: Int?,
+                                  placeholder: String,
+                                  lineNum: Int?,
 ): Int? {
     val groups = warningRegex.find(this)?.groups ?: return null
 
-    val line = lineGroupIdx?.let {
-        try {
-            groups[lineGroupIdx]!!.value.toInt()
-        } catch (e: Exception) {
-            throw ResourceFormatException("Could not extract line number from line [$this], cause: ${e.message}")
+    return lineGroupIdx?.let {
+        groups[lineGroupIdx]!!.value.toIntOrNull() ?: run {
+            val warnLine = groups[lineGroupIdx]!!.value
+            if (warnLine[0] != placeholder[0]) {
+                throw ResourceFormatException("The placeholder in warning must start with a ${placeholder[0]}")
```

Comment:
This case correspond to when group is neither a number nor a placeholder. We should state this in the error message too, what if user accidentally inputs `12o` instead of `120` and you say that it doesn't start with `$`, despite it was intended to be a number?